ios/Classes/OneSignalManager.m

```diff
@@ -0,0 +1,92 @@
+//
+//  OneSignalManager.m
+//  com.williamrijksen.onesignal
+//
+//  Created by William Rijksen on 01-08-17.
+//
+//
+
+#import "OneSignalManager.h"
+#import <OneSignal/OneSignal.h>
+#import "TiApp.h"
+
+@implementation OneSignalManager {}
+
+- (void) receivedHandler:(OSNotification *)notification {
+    OSNotificationPayload* payload = notification.payload;
+    
+    NSString* title = @"";
+    NSString* body = @"";
+    NSDictionary* additionalData = [[NSDictionary alloc] init];
+    
+    if (payload.title) {
+        title = payload.title;
+    }
+    
+    if (payload.body) {
+        body = [payload.body copy];
+    }
+    
+    if (payload.additionalData) {
+        additionalData = payload.additionalData;
+    }
+    
+    NSDictionary* notificationData = @{
+                                       @"title": title,
+                                       @"body": body,
+                                       @"additionalData": additionalData
+                                       };
+    [self.delegate notificationReceived:notificationData];
+}
+
+- (void) actionHandler:(OSNotificationOpenedResult *)result {
+    OSNotificationPayload* payload = result.notification.payload;
+    
+    NSString* title = @"";
+    NSString* body = @"";
+    NSDictionary* additionalData = [[NSDictionary alloc] init];
+    
+    if(payload.title) {
+        title = payload.title;
+    }
+    
+    if (payload.body) {
+        body = [payload.body copy];
+    }
+    
+    if (payload.additionalData) {
+        additionalData = payload.additionalData;
+    }
+    
+    NSDictionary* notificationData = @{
+                                       @"title": title,
+                                       @"body": body,
+                                       @"additionalData": additionalData};
+    [self.delegate notificationReceived:notificationData];
```

Comment:
@williamrijksen a copy and paste issue here... this should be `notificationOpened`